inference-engine/src/vpu/myriad_plugin/myriad_plugin.cpp

```diff
@@ -98,24 +135,128 @@ void Engine::QueryNetwork(
+            }
+        }
 
-    const auto log = std::make_shared<Logger>(
-        "GraphCompiler",
-        parsedConfigCopy.logLevel(),
-        defaultOutput(parsedConfigCopy.compilerLogFilePath()));
+        auto clonedNetwork = CloneAndTransformNetwork(network);
+
+        std::unordered_set<std::string> supported;
+        std::unordered_set<std::string> unsupported;
+
+        std::unordered_set<std::string> splitNames;
+        std::unordered_set<std::string> concatNames;
+
+        std::vector<std::shared_ptr<ngraph::Node>> splits;
+        std::vector<std::shared_ptr<ngraph::Node>> concats;
+
+        for (CNNNetworkIterator itLayer{clonedNetwork.get()};
+             itLayer != CNNNetworkIterator();
+             itLayer++) {
+            auto layerIsSupported = [&] {
+                auto node = (*itLayer)->getNode();
+
+                if (std::dynamic_pointer_cast<const ::ngraph::opset3::Split>(node) != nullptr) {
+                    splitNames.emplace(node->get_friendly_name());
+                    splits.push_back(node);
+                    return false;
+                } else if (std::dynamic_pointer_cast<const ::ngraph::opset3::Concat>(node) != nullptr) {
+                    concatNames.emplace(node->get_friendly_name());
+                    concats.push_back(node);
+                    return false;
+                } else {
+                    auto stageBuilder = std::make_shared<StageBuilder>();
+                    auto frontEnd = std::make_shared<FrontEnd>(stageBuilder, GetCore());
+                    return frontEnd->layerIsSupported((*itLayer)->type);
+                }
+            }();
+
+            const auto fusedNode = (*itLayer)->getNode();
+            if (fusedNode == nullptr) {
+                continue;
+            }
+
+            for (auto&& fusedLayerName : ngraph::getFusedNamesVector(fusedNode)) {
+                if (contains(originalOps, fusedLayerName)) {
+                    if (layerIsSupported) {
+                        supported.emplace(fusedLayerName);
+                    } else {
+                        unsupported.emplace(fusedLayerName);
+                    }
+                }
+            }
+        }
 
-    const auto layerNames = getSupportedLayers(
-        network,
-        static_cast<Platform>(parsedConfigCopy.platform()),
-        parsedConfigCopy.compileConfig(),
-        log,
-        GetCore());
+        for (auto&& layerName : supported) {
+            if (contains(unsupported, layerName)) {
+                supported.erase(layerName);
+            }
+        }
 
-    for (const auto& layerName : layerNames) {
-        res.supportedLayersMap.insert({ layerName, GetName() });
+        unsupported.clear();
+
+        for (const auto& split : splits) {
+            bool is_supported = true;
+            const auto outputs = split->outputs();
+            for (const auto& output : outputs) {
+                const auto& name = output.get_node()->get_friendly_name();
```

Comment:
`output.get_node()` returns `split` node here

Reply:
`get_target_inputs` should be used here. Note that it returns set of target inputs as output can be connected to different node inputs

Reply:
Done